libs/animation/animation.ts

```diff
@@ -1,13 +1,52 @@
+enum AnimationPath {
+    //% block="fly to center of screen"
+    FlyToCenter,
+    //% block="shake"
+    Shake,
+    //% block="bounce (right)"
+    BounceRight,
+    //% block="bounce (left)"
+    BounceLeft,
+    //% block="parachute (right)"
+    ParachuteRight,
+    //% block="parachute (left)"
+    ParachuteLeft,
+    //% block="ease (right)"
+    EaseRight,
+    //% block="ease (left)"
+    EaseLeft,
+    //% block="ease (down)"
+    EaseDown,
+    //% block="ease (up)"
+    EaseUp,
+    //% block="wave (right)"
+    WaveRight,
+    //% block="wave (left)"
+    WaveLeft,
+    //% block="bobbing (in place)"
+    Bobbing,
+    //% block="bobbing (right)"
+    BobbingRight,
+    //% block="bobbing (left)"
+    BobbingLeft
+}
+
 /*
     Animation library for sprites
 */
 //% color="#03AA74" weight=78 icon="\uf021"
 namespace animation {
-    //Handles all the updates
-    let animations: Animation[];
+    export interface AnimationState {
+        movementAnimation?: MovementAnimation,
+        imageAnimation?: ImageAnimation
+    }
+
+    // Stores the animations for the current scene
+    let animations: AnimationState;
 
+    // Preserves animations when switching back and forth between scenes
     let animationStateStack: {
```

Comment:
you can store state in the scene itself: 

    let state = scene.state["__animations"] as YOURDATA